src/App.tsx

```diff
@@ -1,39 +1,59 @@
-import React from 'react';
+import { Component } from 'react';
 import './App.scss';
+import { Clock } from './components/Clock';
 
 function getRandomName(): string {
   const value = Date.now().toString().slice(-4);
 
   return `Clock-${value}`;
 }
 
-export const App: React.FC = () => {
-  const today = new Date();
-  let clockName = 'Clock-0';
-
-  // This code starts a timer
-  const timerId = window.setInterval(() => {
-    clockName = getRandomName();
-  }, 3300);
-
-  // this code stops the timer
-  window.clearInterval(timerId);
-
-  return (
-    <div className="App">
-      <h1>React clock</h1>
-
-      <div className="Clock">
-        <strong className="Clock__name">
-          {clockName}
-        </strong>
-
-        {' time is '}
+interface State {
+  clockName: string;
+  hasClock: boolean;
+  timerId: number;
+}
 
-        <span className="Clock__time">
-          {today.toUTCString().slice(-12, -4)}
-        </span>
+export class App extends Component<{}, State> {
+  state = {
+    clockName: 'Clock-0',
+    hasClock: true,
+    timerId: 0,
+  };
+
+  componentDidMount() {
+    document.addEventListener('contextmenu', this.RightClick);
+    document.addEventListener('click', this.LeftClick);
+
+    this.state.timerId = window.setInterval(() => {
+      this.setState({ clockName: getRandomName() });
+    }, 3300);
+  }
+
+  componentWillUnmount() {
+    document.removeEventListener('contextmenu', this.RightClick);
+    document.removeEventListener('click', this.LeftClick);
+
+    window.clearInterval(this.state.timerId);
+  }
+
+  RightClick = (event: MouseEvent) => {
+    event.preventDefault();
+    this.setState({ hasClock: false });
+  };
+
+  LeftClick = () => {
+    this.setState({ hasClock: true });
+  };
```

Comment:
you should rename names of this methods to something like this: "handleLeftClick" according to checklist:
3. [CODE STYLE] - Method's name should start with a verb (so you could easily tell what your method actually do)

Reply:
good suggestion, thanks)